### ISSUE TITLE
Improve speed of CoroutineExecutorService within StateFragmentLocalTest

### ISSUE THREAD

**Issue by BenHenning** [oppia/oppia-android#1765]
#1764 introduced a CoroutineExecutorService to stabilize tests in #1630. Unfortunately, something about the final implementation of the executor service is particularly inefficient, resulting in tests that previously took 1-2 seconds now taking 11-20 seconds. See: https://github.com/oppia/oppia-android/pull/1630#issuecomment-685427545. While the total test suite time is well below the limit where it would cause issues in CI, this is generally a bad state to be in. Someone will need to dig into why the executor service isn't running optimally, and find ways to optimize it (note that the original implementation didn't have this issue, but it wasn't checked in due to having many other problems that were discovered & fixed as part of writing the initial test suite for the service).

**Comment by BenHenning** [comment]
I suspect that this is in part caused by CoroutineExecutorService performing operations (especially in the context of its test suite) in a way that is not cooperative with cancellability, so the timeout functions aren't actually early exiting as expected.

**Comment by BenHenning** [comment]
It turns out this is even lower priority: StateFragmentLocalTest actually isn't affected after adding a tear-down to flush all tasks from the dispatcher queue (it seems Glide was trying to terminate previous operations that were blocked on the dispatchers wrapping up background work). Keeping this open since the executor service tests do still need to be improved (and likely the implementation of the executor service itself).
